src/monitor/commands/edit-monitor-availability.command.ts

```diff
@@ -0,0 +1,138 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { days } from '../dto/monitor-availability.dto';
+import {
+  EditMonitorBodyMissingFieldsException,
+  EmptyAvailabilityException,
+  MonitoringNotFoundException,
+  UserLoggedNotResponsableForMonitoringException,
+} from '../utils/exceptions';
+import { EditMonitorAvailabilityDTO } from '../dto/edit-monitor-availability.dto';
+
+@Injectable()
+export class EditMonitorAvailabilityCommand {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async execute(
+    monitorId: number,
+    tokenUserId: number,
+    preferentialPlace: string,
+    availability: days[],
+  ): Promise<EditMonitorAvailabilityDTO> {
+    if (!preferentialPlace && !availability)
+      throw new EditMonitorBodyMissingFieldsException();
+
+    if (!availability.length) {
```

Comment:
Pra funcionar como conversamos tem que ser assim:

```suggestion
    if (availability !== undefined && !availability.length) {
```

Reply:
porra vdd, mb. feito